translator/translate/otel/extension/agenthealth/translator.go

```diff
@@ -31,14 +31,25 @@ var (
 )
 
 type translator struct {
-	name               string
-	operations         []string
-	isUsageDataEnabled bool
-	factory            extension.Factory
+	name                string
+	operations          []string
+	isUsageDataEnabled  bool
+	factory             extension.Factory
+	isStatusCodeEnabled bool
 }
 
 var _ common.Translator[component.Config] = (*translator)(nil)
 
+func NewTranslatorWithStatusCode(name component.DataType, operations []string, isStatusCodeEnabled bool) common.Translator[component.Config] {
+	return &translator{
+		name:                name.String(),
+		operations:          operations,
+		factory:             agenthealth.NewFactory(),
+		isUsageDataEnabled:  envconfig.IsUsageDataEnabled(),
+		isStatusCodeEnabled: isStatusCodeEnabled,
+	}
+}
```

Comment:
Why don't we just combine this with the `NewTranslator` function? Is there any case where this function is called when `isStatusCodeEnabled` is false?

Or you can do this instead.
```suggestion
func NewTranslatorWithStatusCode(name component.DataType, operations []string) common.Translator[component.Config] {
	return &translator{
		name:                name.String(),
		operations:          operations,
		factory:             agenthealth.NewFactory(),
		isUsageDataEnabled:  envconfig.IsUsageDataEnabled(),
		isStatusCodeEnabled: true,
	}
}
```